plugins/logging/pkg/backend/metadata.go

```diff
@@ -0,0 +1,269 @@
+package backend
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/opensearch-project/opensearch-go/v2/opensearchutil"
+	opnicorev1 "github.com/rancher/opni/pkg/apis/core/v1"
+	managementv1 "github.com/rancher/opni/pkg/apis/management/v1"
+	"github.com/rancher/opni/pkg/capabilities/wellknown"
+	"github.com/rancher/opni/pkg/opensearch/opensearch/types"
+	"github.com/rancher/opni/pkg/resources/multiclusterrolebinding"
+	"go.uber.org/zap"
+	"golang.org/x/exp/slices"
+	"golang.org/x/sync/errgroup"
+	"io"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"net/http"
+	"os"
+	"time"
+)
+
+func (b *LoggingBackend) waitForOpensearchClient(ctx context.Context) error {
+	var retErr error
+	stopChan := make(chan struct{})
+
+	wait.Until(func() {
+		select {
+		case <-ctx.Done():
+			retErr = fmt.Errorf("context cancelled before client was set")
+			close(stopChan)
+		default:
+		}
+		if b.OpensearchManager.Client != nil {
+			close(stopChan)
+		} else {
+			b.Logger.Errorf("opensearch client not set, waiting")
+		}
+	}, time.Second, stopChan)
+
+	return retErr
+}
+
+func (b *LoggingBackend) reconcileClusterDoc(ctx context.Context, doc *types.ClusterMetadataDoc, cluster *opnicorev1.Cluster) error {
```

Comment:
We should move the logic of this into the opensearch data manager itself.  We could do the wait for client inside of that as well.